front/components/agent_builder/capabilities/knowledge/CapabilitiesConfigurationSheet.tsx

```diff
@@ -0,0 +1,274 @@
+import type { MultiPageSheetPage } from "@dust-tt/sparkle";
+import { MultiPageSheet, MultiPageSheetContent } from "@dust-tt/sparkle";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useEffect, useState } from "react";
+import { useForm, useWatch } from "react-hook-form";
+
+import { useAgentBuilderContext } from "@app/components/agent_builder/AgentBuilderContext";
+import type { AgentBuilderFormData } from "@app/components/agent_builder/AgentBuilderFormContext";
+import { DescriptionSection } from "@app/components/agent_builder/capabilities/knowledge/shared/DescriptionSection";
+import { JsonSchemaSection } from "@app/components/agent_builder/capabilities/knowledge/shared/JsonSchemaSection";
+import {
+  getDataSourceConfigurations,
+  getJsonSchema,
+  getTimeFrame,
+  hasDataSourceSelections,
+  isValidPage,
+} from "@app/components/agent_builder/capabilities/knowledge/shared/sheetUtils";
+import { TimeFrameSection } from "@app/components/agent_builder/capabilities/knowledge/shared/TimeFrameSection";
+import type { CapabilityConfig } from "@app/components/agent_builder/capabilities/knowledge/utils";
+import { CAPABILITY_CONFIGS } from "@app/components/agent_builder/capabilities/knowledge/utils";
+import type {
+  CapabilityFormData,
+  ConfigurationSheetPageId,
+  KnowledgeServerName,
+} from "@app/components/agent_builder/types";
+import type { AgentBuilderAction } from "@app/components/agent_builder/types";
+import {
+  capabilityFormSchema,
+  CONFIGURATION_SHEET_PAGE_IDS,
+} from "@app/components/agent_builder/types";
+import { useSpacesContext } from "@app/components/assistant_builder/contexts/SpacesContext";
+import { DataSourceViewsSpaceSelector } from "@app/components/data_source_view/DataSourceViewsSpaceSelector";
+import { FormProvider } from "@app/components/sparkle/FormProvider";
+import type { DataSourceViewSelectionConfigurations } from "@app/types";
+
+interface CapabilitiesConfigurationSheetProps {
+  capability: KnowledgeServerName | null;
+  onSave: (action: AgentBuilderAction) => void;
+  isOpen: boolean;
+  onClose: () => void;
+  action?: AgentBuilderAction;
+}
+
+export function CapabilitiesConfigurationSheet({
+  capability,
+  onSave,
+  isOpen,
+  onClose,
+  action,
+}: CapabilitiesConfigurationSheetProps) {
+  // We store as state to control the timing to update the content for exit animation.
+  const [config, setConfig] = useState<CapabilityConfig | null>(null);
+
+  const handleClose = () => {
+    onClose();
+
+    // Wait until closing animation ends, otherwise exit animation won't work.
+    setTimeout(() => {
+      setConfig(null);
+    }, 200);
```

Comment:
This is the only hack I can think of enabling the exit animation without using AnimatePresence 🙃 I honestly think it's not too bad to control this by our own as long as we do it sparkle? This will open up the way to mount/remount the content by conditionally rendering with isOpen. I will think about it more but let's discuss when you are back?